### PR TITLE
Use packed structures for networking

### PR DESCRIPTION
**Pull request by d4s**
PXE boot is broken with GCC 7.1 due option '-fstore-merging' enabled
by default for '-O2':

BOOTP broadcast 1
data abort
pc : [<8ff8bb30>]          lr : [<00004f1f>]
reloc pc : [<17832b30>]    lr : [<878abf1f>]
sp : 8f558bc0  ip : 00000000     fp : 8ffef5a4
r10: 8ffed248  r9 : 8f558ee0     r8 : 8ffef594
r7 : 0000000e  r6 : 8ffed700     r5 : 00000000  r4 : 8ffed74e
r3 : 00060101  r2 : 8ffed230     r1 : 8ffed706  r0 : 00000ddd
Flags: nzcv  IRQs off  FIQs off  Mode SVC_32
Resetting CPU ...

Core reason is usage of structures for network headers without packed
attribute.

Signed-off-by: Denis Pynkin <denis.pynkin@collabora.com>

### CODE REVIEW

**Comment by fabioestevam** [comment]
Hi Denis,

On Fri, Jul 21, 2017 at 11:43 AM, Denis Pynkin <notifications@github.com>
wrote:

> PXE boot is broken with GCC 7.1 due option '-fstore-merging' enabled
> by default for '-O2':
>
> BOOTP broadcast 1
> data abort
> pc : [<8ff8bb30>] lr : [<00004f1f>]
> reloc pc : [<17832b30>] lr : [<878abf1f>]
> sp : 8f558bc0 ip : 00000000 fp : 8ffef5a4
> r10: 8ffed248 r9 : 8f558ee0 r8 : 8ffef594
> r7 : 0000000e r6 : 8ffed700 r5 : 00000000 r4 : 8ffed74e
> r3 : 00060101 r2 : 8ffed230 r1 : 8ffed706 r0 : 00000ddd
> Flags: nzcv IRQs off FIQs off Mode SVC_32
> Resetting CPU ...
>
> Core reason is usage of structures for network headers without packed
> attribute.
>
> Signed-off-by: Denis Pynkin denis.pynkin@collabora.com
>

Please send this patch to the U-Boot list.

A similar one has been sent recently:
https://patchwork.ozlabs.org/patch/787298/

Thanks

**Review by ykaliuta — APPROVED** [review]

**Comment by d4s** [comment]
Sent to mail list.

**Comment by otavio** [comment]
@d4s could you test the patch referred by @fabioestevam ? the one which is merged we can backport here.

**Comment by d4s** [comment]
@otavio It is a part of my patch too -- it is necessary (not works without it) but not enough.
For example 'bootp' throw a data error while working here: https://github.com/Freescale/u-boot-fslc/blob/2017.07%2Bfslc/net/bootp.c#L755

PS Sorry, but google did not suggest me appropriate thread in maillist then I'm tried to find the problem.

**Comment by d4s** [comment]
The message to list is still waiting for moderator's approval.

**Comment by otavio** [comment]
It would be great if you could subscribe to the list and resend it.

Often it takes a while for the message to be accepted; I'd like to hear their feedback before committing it.

**Comment by d4s** [comment]
As you suggested I subscribed to list and re-sent my mail after subscription. However for now I have only:
```
Your mail to 'U-Boot' with the subject
    [U-Boot] [PATCH] Use packed structures for networking
Is being held until the list moderator can review it for approval.
The reason it is being held:
    Post to moderated list
```
Even if patch will be rejected, I hope at least description and tests would be useful for further fix.

**Comment by d4s** [comment]
Applied as https://patchwork.ozlabs.org/patch/792238/

**Pull request closed without merging**
